service/controller/resource/remotewritesecret/delete.go

```diff
@@ -0,0 +1,46 @@
+package remotewritesecret
+
+import (
+	"context"
+
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/operatorkit/v7/pkg/controller/context/resourcecanceledcontext"
+
+	"github.com/giantswarm/prometheus-meta-operator/pkg/remotewriteutils"
+)
+
+func (r *Resource) EnsureDeleted(ctx context.Context, obj interface{}) error {
+	r.logger.Debugf(ctx, "deleting prometheus remoteWrite secret")
+	{
+		remoteWrite, err := remotewriteutils.ToRemoteWrite(obj)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+
+		// fetch current prometheus using the selector provided in remoteWrite resource.
+		prometheusList, err := remotewriteutils.FetchPrometheusList(ctx, toResourceWrapper(r), remoteWrite)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+		if prometheusList == nil && len(prometheusList.Items) == 0 {
+			r.logger.Debugf(ctx, "no prometheus found, cancel reconciliation")
+			resourcecanceledcontext.SetCanceled(ctx)
```

Comment:
We should just return `nil` and not cancel reconciliation here.

Canceling reconciliation will leave the finalizer on the `RemoteWrite` resource and it will therefore not be deleted. Implementation details in [operatorkit](https://github.com/giantswarm/operatorkit/blob/dd879b4b4909d2fefe669ea2fbfb6095c8bff845/pkg/controller/controller.go#L439-L478)